diff -Nur moc-2.4.1.org/common.c moc-2.4.1/common.c
--- moc-2.4.1.org/common.c	2006-11-10 10:28:48.000000000 +0100
+++ moc-2.4.1/common.c	2006-11-13 00:42:09.000000000 +0100
@@ -109,7 +109,8 @@
 	char *moc_dir = options_get_str ("MOCDir");
 	
 	if (moc_dir[0] == '~') {
-		if (!(home_dir = getenv("HOME")))
+		if(!(home_dir=getenv("HOME_ETC")))
+		    if (!(home_dir = getenv("HOME")))
 			fatal ("No HOME environmential variable.");
 		if (snprintf(fname, sizeof(fname), "%s/%s/%s", home_dir,
 				(moc_dir[1] == '/') ? moc_dir + 2 : moc_dir + 1,
