diff -ur moc-1.1.0-orig/configure.in moc-1.1.0/configure.in
--- moc-1.1.0-orig/configure.in	Tue Dec  3 18:25:40 2002
+++ moc-1.1.0/configure.in	Mon Feb  3 22:14:31 2003
@@ -47,8 +47,8 @@
 
 dnl ncurses
 AC_CHECK_LIB(ncurses, initscr, [
-	AC_CHECK_HEADER([ncurses.h],, [
-		AC_MSG_ERROR([You nedd ncurses header file.])])
+	AC_CHECK_HEADER([ncurses/ncurses.h],, [
+		AC_MSG_ERROR([You need ncurses header file.])])
 	],
 	[AC_MSG_ERROR([You need ncurses library.])]
 )
