--- moc-2.4.3/decoder_plugins/ffmpeg/ffmpeg.c.orig	2008-10-13 00:00:40.792474789 +0200
+++ moc-2.4.3/decoder_plugins/ffmpeg/ffmpeg.c	2008-10-13 00:01:03.290928519 +0200
@@ -22,7 +22,7 @@
 #ifdef HAVE_INTTYPES_H
 # include <inttypes.h>
 #endif
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 
 /* FFmpeg also likes common names, without that, our common.h and log.h would
  * not be included. */
